tests/testFlagHandler.py

```diff
@@ -38,49 +39,98 @@
+                ("flag_containsNan","Measurement area contains a nan"),
+                ("flag_edge", "Measurement area over edge"))
 class PythonPlugin(SingleFramePlugin):
     '''
-    This is a sample Python plugin.  The flag handler for this plugin is created
-    during construction, and is called using the method fail().  All plugins are
-    required to implement this method, which is used to set the flags in the
-    output source record if an error occurs.
+    This is a sample Python plugin which shows how to create and use a FlagHandler.
+    The FlagHandler defines the known failures which can occur when the plugin is
+    called, and should be tested after measure() to detect any potential problems.
+
+    This plugin is a very simple-minded flux measurement algorithm which sums
+    the pixel values in a square box of dimension config.size around the center point.
+
+    The flagHandler for this plugin is created during construction as a result of the
+    addFlagHandler decorator, which adds a general failure flag and two specific flags
+    (containsNan and edge) to the schema. The class variable ErrEnum is also added.
+    This is an enumeration of the 3 flags.
+
+    Note that to properly set the error flags when a MeasurementError occurs, the plugin
+    must implement the fail() method as shown below. The fail method should set both
+    the general error flag, and any specific flag (such as ErrEnum.flag_edge or
+    ErrEnum.flag_containsNan) as designated in the MeasurementError.
+
+    This example also demonstrates the use of the SafeCentroidExtractor. The
+    SafeCentroidEextractor and SafeShapeExtractor can be used to get some reasonable
+    estimate of the centroid or shape in cases where the centroid or shape slot
+    has failed on a particular source.
     '''
+
     ConfigClass = PythonPluginConfig
-    # Class variables ErrEnum and FLAGDEFS are added by the decorator
 
     @classmethod
     def getExecutionOrder(cls):
-        return cls.SHAPE_ORDER
+        return cls.FLUX_ORDER
 
+    #   Note that because of the decorator, a flagHander and ErrEnum are also created
+    #   and the flags are added to the schema.
     def __init__(self, config, name, schema, metadata):
         SingleFramePlugin.__init__(self, config, name, schema, metadata)
-        # The instance variable flagHandler is added by the decorator
+        self.centroidExtractor = lsst.meas.base.SafeCentroidExtractor(schema, name)
+        self.fluxKey = schema.addField(name + "_flux", "F", doc = "flux")
+        self.magKey = schema.addField(name + "_mag", "F", doc = "mag")
 
-    #   This is a measure routine which does nothing except to raise Exceptions
-    #   as requested by the caller. Errors normally don't occur unless there is
-    #   something wrong in the inputs, or if there is an error during the measurement
+    #   The measure method is called by the measurement framework when task.run is called
```

Comment:
Normally Python projects don't write doc strings in unit tests, but since this code is doubling as an example, and features comments that *look* like doc strings, I thought I might as well encourage the python way of documenting a class, function, or method, which is a docstring.